packages/talos_messenger_core/src/core.rs

```diff
@@ -0,0 +1,46 @@
+use ahash::HashMap;
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use strum::{Display, EnumIter, EnumString};
+
+use crate::errors::MessengerServiceResult;
+
+#[derive(Debug, Display, Serialize, Deserialize, EnumString, EnumIter, Clone, Eq, PartialEq)]
+pub enum CommitActionType {
+    #[strum(serialize = "publish")]
+    Publish,
+}
+
+#[derive(Debug, Display, Serialize, Deserialize, EnumString, Clone, Eq, PartialEq)]
+pub enum PublishActionType {
```

Comment:
How about we rename this to represent a list of brokers? If I read the intention correctly this enum wants to list what messaging brokers we can publish to. In that case its no so much of a "...ActionType" anymore.

Reply:
Actually can I leave this comment for the next PR please? Some part of it actually would get more generic. Like `line 10 - CommitActionType` is not used anymore.  And my idea around the implementation has become bit more generic since the time these enums where created.